pyracf/common/security_admin.py

```diff
@@ -219,6 +211,15 @@ def _clear_state(self) -> None:
         self._segment_traits = {}
         self._trait_map = {}
 
+    def _reset_state(self) -> None:
+        """Reset state once requests are done."""
+        del self._segment_traits
+        del self._trait_map
+        del self.__preserved_segment_traits
```

Comment:
I am not sure if deleting these values is necessary. I think it should be ok to just reset them to empty dictionary.

Reply:
I am guessing that the del might be a security thing maybe. Is the purpose to remove this data completely from memory instead of just changing the references to ensure that no sensitive data is accessible in memory after the request is complete?

Reply:
That was the intention. With python's memory management, I'm not certain that it's really any better than setting them to empty, but it seemed like it was the more security-conscious thing to at least try to do.